File: 2-ad-hoc/1267.cpp
#include<iostream>
//
int**matrix,x,y;
int check(){
	for(int i=0;i<x;++i){
		int temp=1;
		for(int j=0;j<y;++j)temp=(!matrix[j][i])?0:temp;
		if(temp)return 1;
	}
	return 0;
}
//
int main(){
	while(std::cin>>x>>y&&x&&y){
		matrix=(int**)malloc(y*sizeof(int*));
		for(int i=0;i<y;++i){
			matrix[i]=(int*)malloc(x*sizeof(int));
			for(int j=0;j<x;++j){
				int temp;
				std::cin>>temp;
				matrix[i][j]=temp;
			}
		}
		if(check())std::cout<<"yes\n";
		else std::cout<<"no\n";
		free(matrix);
	}
	return 0;
}

File: 2-ad-hoc/1267.py
matrix,x,y=None,None,None
def check():
    for i in range(x):
        temp=1
        for j in range(y):
            temp=matrix[j][i] if temp else 0
        if temp:return 1
    return 0
while(1):
    [x,y]=list(map(int,input().split()))
    if not x and not y:break
    matrix=[[0 for j in range(x)] for i in range(y)]
    for i in range(y):
        index=0
        for j in input().split():
            matrix[i][index]=int(j)
            index+=1
    print('yes') if check() else print('no')
    matrix=None
